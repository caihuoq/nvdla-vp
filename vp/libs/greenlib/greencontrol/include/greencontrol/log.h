//   GreenControl framework
//
// LICENSETEXT
//
//   Copyright (C) 2009 : GreenSocs Ltd
// 	 http://www.greensocs.com/ , email: info@greensocs.com
//
//   Developed by :
//   
//   Michael Ruetz <m.ruetz@tu-bs.de>,
//   Christian Schroeder <schroeder@eis.cs.tu-bs.de>
//     Technical University of Braunschweig, Dept. E.I.S.
//     http://www.eis.cs.tu-bs.de
//
//
//   This program is free software.
// 
//   If you have no applicable agreement with GreenSocs Ltd, this software
//   is licensed to you, and you can redistribute it and/or modify
//   it under the terms of the GNU General Public License as published by
//   the Free Software Foundation; either version 2 of the License, or
//   (at your option) any later version.
// 
//   If you have a applicable agreement with GreenSocs Ltd, the terms of that
//   agreement prevail.
// 
//   This program is distributed in the hope that it will be useful,
//   but WITHOUT ANY WARRANTY; without even the implied warranty of
//   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//   GNU General Public License for more details.
// 
//   You should have received a copy of the GNU General Public License
//   along with this program; if not, write to the Free Software
//   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
//   02110-1301  USA 
// 
// ENDLICENSETEXT

// doxygen comments

#ifndef __LOG_H__
#define __LOG_H__

#include "greencontrol/gclog/gc_logger.h"
#include "greencontrol/gclog/filter.h"
#include "greencontrol/gclog/output.h"

#endif
