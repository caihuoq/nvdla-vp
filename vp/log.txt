-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for TLM
running ls /usr/local/systemc-2.3.0/include/tlm.h 2>&1
/usr/local/systemc-2.3.0/include/tlm.h
-- TLM library = /usr/local/systemc-2.3.0/include/tlm.h
-- Searching for NVDLA CMOD
-- NVDLA HW VERSION = nvdla_os_initial
-- NVDLA CMOD include directory = /usr/local/nvdla/hw/outdir/nv_full/cmod/release/include
-- NVDLA CMOD library directory = /usr/local/nvdla/hw/outdir/nv_full/cmod/release/lib
-- NVDLA CMOD library = /usr/local/nvdla/hw/outdir/nv_full/cmod/release/lib/libnvdla_cmod.so
-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for TLM
running ls /usr/local/systemc-2.3.0/include/tlm.h 2>&1
/usr/local/systemc-2.3.0/include/tlm.h
-- TLM library = /usr/local/systemc-2.3.0/include/tlm.h
-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for TLM
running ls /usr/local/systemc-2.3.0/include/tlm.h 2>&1
/usr/local/systemc-2.3.0/include/tlm.h
-- TLM library = /usr/local/systemc-2.3.0/include/tlm.h
-- Searching for GreenLib
-- GreenLib include directory = /workspace/vp/libs/greenlib/greencontrol/include
-- GreenLib library directory = /workspace/vp/libs/greenlib/greenreg/lib
-- Searching for TLM2C
-- TLM2C include directory = /workspace/vp/libs/tlm2c/include
-- TLM2C library directory = /workspace/vp/libs/tlm2c.build
-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for TLM
running ls /usr/local/systemc-2.3.0/include/tlm.h 2>&1
/usr/local/systemc-2.3.0/include/tlm.h
-- TLM library = /usr/local/systemc-2.3.0/include/tlm.h
-- Searching for GreenLib
-- GreenLib include directory = /workspace/vp/libs/greenlib/greencontrol/include
-- GreenLib library directory = /workspace/vp/libs/greenlib/greenreg/lib
-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for TLM
running ls /usr/local/systemc-2.3.0/include/tlm.h 2>&1
/usr/local/systemc-2.3.0/include/tlm.h
-- TLM library = /usr/local/systemc-2.3.0/include/tlm.h
-- Searching for GreenLib
-- GreenLib include directory = /workspace/vp/libs/greenlib/greencontrol/include
-- GreenLib library directory = /workspace/vp/libs/greenlib/greenreg/lib
-- Searching for NVDLA CMOD
-- NVDLA HW VERSION = nvdla_os_initial
-- NVDLA CMOD include directory = /usr/local/nvdla/hw/outdir/nv_full/cmod/release/include
-- NVDLA CMOD library directory = /usr/local/nvdla/hw/outdir/nv_full/cmod/release/lib
-- NVDLA CMOD library = /usr/local/nvdla/hw/outdir/nv_full/cmod/release/lib/libnvdla_cmod.so
-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for TLM
running ls /usr/local/systemc-2.3.0/include/tlm.h 2>&1
/usr/local/systemc-2.3.0/include/tlm.h
-- TLM library = /usr/local/systemc-2.3.0/include/tlm.h
-- Searching for GreenLib
-- GreenLib include directory = /workspace/vp/libs/greenlib/greencontrol/include
-- GreenLib library directory = /workspace/vp/libs/greenlib/greenreg/lib
-- Searching for SimpleCPU model.
-- SimpleCPU include directory = /workspace/vp/models/cpu/include
-- SimpleCPU library directory = /workspace/vp/models/cpu.build
-- Configuring done
-- Generating done
-- Build files have been written to: /workspace/vp
