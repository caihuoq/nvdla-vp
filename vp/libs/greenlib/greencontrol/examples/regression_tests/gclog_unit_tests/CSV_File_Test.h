//   GreenControl framework
//
// LICENSETEXT
//
//   Copyright (C) 2009 : GreenSocs Ltd
// 	 http://www.greensocs.com/ , email: info@greensocs.com
//
//   Developed by :
//   
//   Michael Ruetz <m.ruetz@tu-bs.de>,
//   Christian Schroeder <schroeder@eis.cs.tu-bs.de>
//     Technical University of Braunschweig, Dept. E.I.S.
//     http://www.eis.cs.tu-bs.de
//
//
// The contents of this file are subject to the licensing terms specified
// in the file LICENSE. Please consult this file for restrictions and
// limitations that may apply.
// 
// ENDLICENSETEXT

// doxygen comments

#ifndef __CSV_FILE_TEST_H__
#define __CSV_FILE_TEST_H__

#include "greencontrol/log.h"
#include "TestCase.h"

// GCLOG_RETURN should be enabled
class CSV_File_Test : public TestCase
{
public:
  CSV_File_Test() {}
  virtual ~CSV_File_Test(void) {}

  std::string identify()
  {
    return "CSV_File_Test";
  }

  void setup()
  {
    pLog = new GC_Logger();
    pCSVOut = new CSV_File("CSV_File_Test.csv", false);
  }

  void destroy()
  {
    delete pLog;
    delete pCSVOut;
  }

  // This is for testing the CSV_File and OutputBase classes.
  void run()
  {
    ControlTransactionHandle ct;
    LogConfig cfg, cfg2;

    // test the generateUniqueFilename() method in OutputBase
    CSV_File noName;
    std::cout << "\nA warning about an invalid filename should follow:\n";
    CSV_File nullName(NULL, true);
    CSV_File uniqueNoEnding("csvNoEnding", true);
    CSV_File uniqueEnding("csvEnding.csv", true);

    ct = getCT();
    cfg.setAnyPointer2(false);
    cfg.setLogPointer(false);

    // use uniqueEnding file for some settings
    uniqueEnding.setDelimiter(",");
    uniqueEnding.enableTiming(false);
    uniqueEnding.setHumanReadable(false);
    // use a static LogConfig -> writeStaticTransaction() is used
    uniqueEnding.setLogConfig(&cfg);
    uniqueEnding.processTransaction(ct);
    // use normal delimiter again
    uniqueEnding.setDelimiter(";");
    ct->set_mMetaData("The header is written after the filter was added.");
    uniqueEnding.wasAdded();
    uniqueEnding.processTransaction(ct);
    ct->set_mMetaData("But it must not be written more than once.");
    uniqueEnding.wasAdded();
    uniqueEnding.processTransaction(ct);

    // use different settings
    ct = getCT();
    cfg2.setAnyPointer(false);
    cfg2.setLogPointer(false);
    cfg2.setCmdIf(false);
    pCSVOut->enableTiming(true);
    pCSVOut->setHumanReadable(true);
    pCSVOut->wasAdded();
    pCSVOut->processTransaction(ct);
    // use a LogConfig in the CT -> writeDynamicTransaction() is used
    ct->set_mMetaData("A LogConfig object is used in the CT.");
    ct->activateConfig(&cfg2);
    pCSVOut->processTransaction(ct);
    // use an unmodified LogConfig that was set to true -> writeFullTransaction() is used
    ct->set_mMetaData("The LogConfig is reset to true.");
    cfg2.reset(true);
    pCSVOut->processTransaction(ct);
    ct->deactivateConfig();
  }

protected:
  GC_Logger* pLog;
  CSV_File* pCSVOut;
};

#endif
