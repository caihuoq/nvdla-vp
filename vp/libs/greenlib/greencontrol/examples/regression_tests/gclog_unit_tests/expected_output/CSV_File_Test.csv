Tuesday, 16 February 2010 - 13:06:06
CSV output used with GC_Logger.


"Time / Delta";"Transaction:";Service;"Service Name";Target;Command;"Command Name";AnyPointer;AnyPointer2;AnyUint;Specifier;Value;ID;"Sender Name";MetaData;Error;LogPointer;"LogPointer Content";CommandIf;LogConfig
"0 s/0";returning;0;"NO_SERVICE";0;0;"Warning: API/Plugin doesn't implement command_if!";0;0;1768763237;"";"";0x93a16d0;"Warning: API/Plugin doesn't implement command_if!";"";0;0;;0;0x93a8468
"0 s/0";returning;0;"NO_SERVICE";0;0;"Warning: API/Plugin doesn't implement command_if!";;0;1768763237;"";"";0x93a16d0;"Warning: API/Plugin doesn't implement command_if!";"A LogConfig object is used in the CT.";0;;;;0xbfe8d400
"0 s/0";returning;0;"NO_SERVICE";0;0;"Warning: API/Plugin doesn't implement command_if!";0;0;1768763237;"";"";0x93a16d0;"Warning: API/Plugin doesn't implement command_if!";"The LogConfig is reset to true.";0;0;;0;0xbfe8d400


Regular end of log.
Finished on Tuesday, 16 February 2010 - 13:06:06